Update MLFlow script

juliafalcao opened the pull request:
## Pull Request Checklist

- [x] The title of the PR is clear and concise.
- [x] I have provided a detailed description of the changes and the reason for the changes.
- [x] I have linked the related issue(s) in the description (if applicable).
- [x] I have followed the coding guidelines of this project.
- [x] I have ensured there are no breaking changes.
- [x] I have updated related documentation (if applicable).
- [x] I have added reviewers, including at least one member of the MLOps team (@hrosegalb, @PaulNdrei, @ankush13r, @igorktech)

## Description

This PR updates the `mlflow_log_results.py` script to include new tasks and update some information:

* Add `esbbq` task and subtasks
* Remove outdated `veritasqa` records
* Add `veritasqa` `gen` and `mc` subtasks in 4 languages
* Add hardcoded random baselines for `veritasqa` because of varying number of labels
* Rename `xnli_es` to `xnli_es_spanish_bench` to use our customized version

## Issue Link(s)
n/a

## Testing

I have not tested this change - I don't really use MLFlow so I'm not sure how to test it at all.

hrosegalb commented on lm_eval/mlflow_log_results.py:
```diff
@@ -1251,6 +1288,8 @@ def get_random(task, num_labels) -> float:
         return 8.33
     elif num_labels in ["gen_task", "2 (mcc)"]:
         return 0.0
+    elif num_labels in ["varying"]:
```
This doesn't really need to be a list, since you're only comparing `num_labels` to one value.

hrosegalb commented on lm_eval/mlflow_log_results.py:
```diff
@@ -986,6 +1022,18 @@
         "language": "gl",
     },
+    "veritasqa_gen_gl": {
+        "num_labels":" gen_task",
```
There is an extra space here that needs to be removed.

hrosegalb reviewed (changes requested):
If I'm not mistaken, we also need to update the dictionary to include `cocoteros_es`, `bbq_en`, and update `xnli_en` to xnli_en_iberobench`.

Tomorrow, I can work on testing your changes and let you know if anything is broken.

juliafalcao commented:
Thank you @hrosegalb for pointing out the errors!

* I updated `xnli_en` to `xnli_en_iberobench`
* I removed the extra spaces
* I added the `cocoteros_es` task
* I did _not_ add `bbq_en` because in the end we are not adding it to IberoBench and we are only keeping this task for internal usage so we don't need to log the results to MLFlow for now.
* I changed the check of the `varying` condition to `==` instead of `in list`.

hrosegalb commented:
MLFlow test results [here](https://eval.mlflow.pro.aina.bsc.es/#/experiments/641841037182948853?searchFilter=&orderByKey=attributes.start_time&orderByAsc=false&startTime=ALL&lifecycleFilter=Active&modelVersionFilter=All+Runs&datasetsFilter=W10%3D)
